### PR TITLE
Import Solana private key

### PR DESCRIPTION
**Pull request by darkdh**
Resolves https://github.com/brave/brave-browser/issues/19475

1. ImportAccount now requires CoinType
2. Add importing Solana private key support (without UI hookup)
3. KeyringServiceUnitTest cleanup

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by bbondy — APPROVED** [review]

**Comment by darkdh** [comment]
this is the only iOS file changed in this PR
`ios/browser/api/brave_wallet/BUILD.gn`

File: components/brave_wallet/browser/BUILD.gn [hunk 1]
```diff
@@ -130,6 +129,21 @@ static_library("ethereum_permission_utils") {
...
+    "brave_wallet_constants.cc",
+    "brave_wallet_constants.h",
+  ]
+
+  deps = [
+    "//base",
+    "//brave/components/brave_wallet/common",
+    "//brave/components/brave_wallet/common:mojom",
+    "//brave/components/resources:strings_grit",
+    "//ui/base:base",
```
Comment (kylehickinson) [inline components/brave_wallet/browser/BUILD.gn:143]: not sure if this is used anywhere (I don't see it used in `brave_wallet_constants.cc`, maybe its not needed?), but careful as many of the files in `//ui/base` are not available in iOS builds

**Review by kylehickinson — APPROVED** [review]
iOS++, just keep in mind above comment 🙂

File: components/brave_wallet/browser/BUILD.gn (hunk 1 above)
Comment (darkdh) [reply components/brave_wallet/browser/BUILD.gn:143]: and that dep `//ui/base` is already there when iOS depends on `brave/components/brave_wallet/browser`

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/AddAccountActivity.java [hunk 2]
```diff
@@ -144,7 +144,7 @@ public void onClick(View v) {
                                     });
                         } else {
                             mKeyringService.importAccount(mAddAccountText.getText().toString(),
-                                    mPrivateKeyControl.getText().toString().trim(),
+                                    mPrivateKeyControl.getText().toString().trim(), CoinType.ETH,
```
Comment (darkdh) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/AddAccountActivity.java:147]: cc @SergeyZhukovsky

File: components/brave_wallet/browser/BUILD.gn (hunk 1 above)
Comment (kylehickinson) [reply components/brave_wallet/browser/BUILD.gn:143]: @darkdh Sounds good, didn't realize it was already part of it :)

**Comment by darkdh** [comment]
`BraveShieldsAPIBrowserTest.AllowScriptsOnceIframe` failure is not related to this PR
https://ci.brave.com/job/pr-brave-browser-import-solana-key-macos/6/execution/node/285/log/

**Pull request merged**